src/unspent_store.py

```diff
@@ -0,0 +1,204 @@
+        fees_coin: Unspent = Unspent(block.body.fees_coin, block.height, 0, 0, 1) # type: ignore # noqa
+        await self.add_unspent(coinbase)
+        await self.add_unspent(fees_coin)
+
+    def nuke_diffs(self):
+        self.head_diffs = dict()
+
+    # Received new tip, just update diffs
+    async def new_heads(self, blocks: List[FullBlock]):
+        last: FullBlock = blocks[-1]
+        diff_store: DiffStore = await DiffStore.create(last.header_block, dict())
+
+        block: FullBlock
+        for block in blocks:
+            removals, additions = await block.tx_removals_and_additions()
+            await self.add_diffs(removals, additions, block, diff_store)
+
+        self.head_diffs[last.header_hash] = diff_store
+
+    async def add_diffs(self, removals: List[bytes32], additions: List[Coin],
+                        block: FullBlock, diff_store: DiffStore):
+
+        for coin_name in removals:
+            removed: Optional[Unspent] = None
+            if coin_name.hex() in diff_store.diffs:
+                removed = diff_store.diffs[coin_name.hex()]
+            if removed is None:
+                removed = await self.get_unspent(coin_name)
+            if removed is None:
+                raise Exception
+            spent = Unspent(removed.coin, removed.confirmed_block_index,
+                            block.height, 1, removed.coinbase) # type: ignore # noqa
+            diff_store.diffs[spent.name.hex()] = spent
+
+        for coin in additions:
+            added: Unspent = Unspent(coin, block.height, 0, 0, 0) # type: ignore # noqa
+            diff_store.diffs[added.name.hex()] = added
+
+        coinbase: Unspent = Unspent(block.body.coinbase, block.height, 0, 0, 1) # type: ignore # noqa
+        diff_store.diffs[coinbase.name.hex()] = coinbase
+        fees_coin: Unspent = Unspent(block.body.fees_coin, block.height, 0, 0, 1) # type: ignore # noqa
+        diff_store.diffs[fees_coin.name.hex()] = fees_coin
+
+    # Store unspent in DB and ram cache
+    async def add_unspent(self, unspent: Unspent) -> None:
+        cursor = await self.unspent_db.execute(
+            "INSERT OR REPLACE INTO unspent VALUES(?, ?, ?, ?, ?, ?)",
+            (unspent.coin.name().hex(),
+             unspent.confirmed_block_index,
+             unspent.spent_block_index,
+             int(unspent.spent),
+             int(unspent.coinbase),
+             bytes(unspent)),
+        )
+        await cursor.close()
+        await self.unspent_db.commit()
+        self.lca_unspent_coins[unspent.coin.name().hex()] = unspent
+        if len(self.lca_unspent_coins) > 600000:
+            while len(self.lca_unspent_coins) > 600000:
+                first_in = list(self.lca_unspent_coins.keys())[0]
+                del self.lca_unspent_coins[first_in]
+
+    # Update unspent to be spent in DB
+    async def set_spent(self, coin_name: bytes32, index: uint32):
```

Comment:
This can be more efficiently done by just sql, change the index and set set spent. This saves a lot of slow serialization and parsing time. However, we should not change the index if it is already spend right? Maybe we should assert that it's not spent.